cmd/event/rm/rm.go

```diff
@@ -61,11 +87,13 @@ func deleteForDevice(dc coredata.EventClient) (err error) {
 }
 
 func validate(args []string) error {
-	if len(args) == 0 && device == "" {
-		return errors.New("no event id/device name provided")
+	if len(args) == 0 && device == "" && age == "" {
+		return errors.New("no event id/device name nor age provided")
 	}
-	if len(args) != 0 && device != "" {
-		return errors.New("provide either event id or device name")
+	if len(args) != 0 && (device != "" || age != "") ||
```

Comment:
Would it be easier for the code maintenance if we listed valid combinations versus invalid? :-)
Including add a comment if something feels logical but not supported currently in EdgeX


Would you please add some comments
Delete events by ID(s)
Delete events associated with a Device given its name
??? If age also provided then events associated with named device older than age
if only age provided, then delete all events (regardless of associated device) older than age

Reply:
What if the client provide ./edgex-cli event rm <id1>, <id2> --device <my_device> --age 1 --unit d
Currently, we do not support mixing different criteria.We strictly follows the provided by EdgeX endpoints. 
But this is an idea we could keep in mind for future enhancement. 
We should think if it is desirable to first find the common section between the provided criteria, or execute them independently  (example: remove all events older than one day that belongs to specific device or 
remove all events older that one day and events that belongs to specific device (independently)) 
we could implement something like parsing "search queries" like in mysql "delete all events where devicename=x and/or age >0" .. event support parenthesis. Of course it won't be easy for implementation ..  
We should research if other cli tool support such more comprehensive criteria